Tests/UnitTests/Purchasing/Purchases/PurchasesPurchasingTests.swift

```diff
@@ -334,13 +334,13 @@ class PurchasesPurchasingTests: BasePurchasesTests {
 
         let product = StoreProduct(sk1Product: MockSK1Product(mockProductIdentifier: "com.product.id1"))
 
+        var result: PurchaseResultData?
         var receivedError: NSError?
 
-        _ = Task {
+        // Need to do this async so the code below can invoke the `updatedTransaction` delegate method.
+        _ = Task<Void, Never> {
             do {
-                _ = try await self.purchases.purchase(
-                    product: product
-                )
+                result = try await self.purchases.purchase(product: product)
             } catch {
                 receivedError = error as NSError
```

Comment:
Could we replace this with a `fail`?

Reply:
No because it's inside of a `Task` so it's happening asynchronously.
There's a comment above explaining why it has to be this way:
> // Need to do this async so the code below can invoke the `updatedTransaction` delegate method.